filecoin/errors.js

```diff
@@ -0,0 +1,38 @@
+export const DatabaseOperationErrorName = /** @type {const} */ (
+  'DatabaseOperationFailed'
+)
+export class DatabaseOperationFailed extends Error {
+  get reason() {
+    return this.message
```

Comment:
Is this necessary? Why not just use the message property?

Reply:
It follows the same pattern as `ucanto` does https://github.com/web3-storage/ucanto/blob/c8999a59852b61549d163532a83bac62290b629d/packages/core/src/result.js#L63 for failures, so that the types of failures from handlers of ucanto server are all well aligned. So, not strictly necessary, but consistent